vimeo/VimeoUpload
# Updates VimeoUpload to support Circle CI 2.0

### Ticket
Based on the VimeoNetworking ticket, [AF-306](https://vimean.atlassian.net/browse/AF-306)

### Ticket Summary
Research and implement the changes required to make the move from Circle CI 1.0 to 2.0 before version 1.0 is sunset on August 31st.

### Implementation Summary
- A new Circle CI config has been added at `.circleci/config.yml`.
- The old config, `circle.yml`, has been removed from the project.

One of the biggest changes from the old config is that the build can be broken down into workflows, and workflows can be broken down into jobs. This PR only introduces one workflow, with just a single job called `build-and-test`.

The new config is much more verbose than the old one. From top to bottom here's an overview of what's going on:

**Jobs**
- Indicate the Circle CI version to use. `2` is required.
- Define the job called "build-and-test".
- Set up environment variables, set the Xcode version, the working directory,  override the shell parameter by passing the `--login` option and enable `pipefail`. _Note that we're overriding the shell parameter here to use the `--login` option, although this may not be strictly necessary._
- Begin the job's steps by checking out the project.
- Restore cached gems, if any.
- Set the Ruby version to match the version we use locally.
- Run `bundle install` and attempt to use cached gems.
- Build and run unit tests for VimeoUpload-iOS via Fastlane.
- Build and run unit tests for VimeoUpload-iOS-OldUpload via Fastlane.
- Cache gems using the prefix `v1-gems-` followed by the `Gemfile.lock`'s checksum.
- Store artifacts, if any.
- Store test results, if any.

**Workflows**
- Define worlflows.
- Indicate the Circle CI version to use. `2` is required.
- List the jobs to run.
- Run the `build-and-test` job.

### How to Test
The build job for this branch can be found [here](https://circleci.com/gh/vimeo/VimeoUpload/tree/circle-ci-upgrade). Builds should be passing.

The original VimeoNetworking PR integrating the new config can be found here: https://github.com/vimeo/VimeoNetworking/pull/257
